prow/cmd/build/controller.go

```diff
@@ -320,24 +321,36 @@ func (c *controller) getBuild(context, namespace, name string) (*buildv1alpha1.B
 	if !ok {
 		return nil, errors.New("context not found")
 	}
-	return b.informer.Lister().Builds(namespace).Get(name)
+	item, _, err := b.informer.GetStore().GetByKey(namespace + "/" + name)
```

Comment:
This is a bit weird and now how it is usually done with controller-runtime. Usually, there is a [`Manager`](https://github.com/kubernetes-sigs/controller-runtime/blob/477bf4f046c31c351b46fa00262bc814ac0bbca1/pkg/manager/manager.go#L45) which manages all controllers and other things that should run as part of the binary as [`Runnable`](https://github.com/kubernetes-sigs/controller-runtime/blob/477bf4f046c31c351b46fa00262bc814ac0bbca1/pkg/manager/manager.go#L146).

The manager then also has a [`GetClient`](https://github.com/kubernetes-sigs/controller-runtime/blob/477bf4f046c31c351b46fa00262bc814ac0bbca1/pkg/manager/manager.go#L69) func which returns a client that by default has `Reader` interface that is cache-backed, meaning all read requests are redirected to a lister:  https://github.com/kubernetes-sigs/controller-runtime/blob/477bf4f046c31c351b46fa00262bc814ac0bbca1/pkg/manager/manager.go#L237

The problem is that it is assumed there is _one_ cluster to talk to and we have multiple here. The approach in itself is IMHO awesome thought, because:
* In your controller, you don't have to explicitly decide if you use the lister or get it from the API
* Writing tests is much easier, as you just need to inject a fake client and not create a fake-client backed lister

The reason why we can't directly use the same `Client` implementation is that parts of it are not exported. Alternatively to what I did in this PR so far, we could just copy the unexported code over. WDYT?

Reply:
Why not a Manager per cluster?

Reply:
Interesting idea, that could work, I'll try it out when I find some time

Reply:
Yup, works like a charm